services/web/server/src/simcore_service_webserver/projects/db.py

```diff
@@ -390,28 +410,40 @@ async def list_projects(  # pylint: disable=too-many-arguments
                         if not include_hidden
                         else sa.text("")
                     )
-                    & (
-                        (projects.c.prj_owner == user_id)
-                        | sa.text(
-                            f"jsonb_exists_any(access_rights_subquery.access_rights, {assemble_array_groups(user_groups)})"
-                        )
-                    )
                     & (
                         (projects_to_products.c.product_name == product_name)
                         # This was added for backward compatibility, including old projects not in the projects_to_products table.
                         | (projects_to_products.c.product_name.is_(None))
                     )
+                    & (
+                        projects_to_folders.c.folder_id == folder_id
+                        if folder_id
+                        else projects_to_folders.c.folder_id.is_(None)
+                    )
+                    & (
+                        projects.c.workspace_id == workspace_id  # <-- Shared workspace
+                        if workspace_id
```

Comment:
also here

Reply:
the same thing, either positive integer or None